src/report.rs

```diff
@@ -486,3 +602,43 @@ impl fmt::Display for CrashReport {
         write!(f, "{}", report.trim())
     }
 }
+
+/// Deduplicate crash reports
+///
+/// # Arguments
+///
+/// * `casreps` - slice of Casr reports
+///
+/// # Return value
+///
+/// An vector of the same length as `casreps`.
+/// Vec[i] is false, if original casrep i is a duplicate of any element of `casreps`.
+pub fn dedup_reports(casreps: &[CrashReport]) -> Result<Vec<bool>> {
+    let mut traces: Vec<Stacktrace> = Vec::new();
+    (0..casreps.len()).into_iter().try_for_each(|i| {
```

Comment:
We may use map and generate the resulting vector

Reply:
The same holds for cluster_reports

Reply:
done